lib/edge_impulse/CMakeLists.txt

```
@@ -20,9 +20,15 @@ if(CONFIG_EDGE_IMPULSE_URI STREQUAL "")
   message(FATAL_ERROR "CONFIG_EDGE_IMPULSE_URI must be specified")
 endif()
 
-# Verify if path is relative to application source directory
-if(EXISTS ${APPLICATION_SOURCE_DIR}/${CONFIG_EDGE_IMPULSE_URI})
-  set(EI_URI ${APPLICATION_SOURCE_DIR}/${CONFIG_EDGE_IMPULSE_URI})
+string(FIND ${CONFIG_EDGE_IMPULSE_URI} "://" IDX)
+
+if(IDX EQUAL -1)
```

I believe we should just check if user is providing `<scheme>://` URL, and if so, don't manipulate it.
```suggestion
set(EI_URI_PATH ${CONFIG_EDGE_IMPULSE_URI})

if(NOT ${EI_URI_PATH} MATCHES "^[a-z]+://")
  if(NOT IS_ABSOLUTE ${EI_URI_PATH})
    set(EI_URI_PATH ${APPLICATION_SOURCE_DIR}/${EI_URI_PATH})
  endif()
endif()
```

but regarding the treatment of relative path, I would like that we conclude on the MCUBoot signature file issue first.

Cause having this setting relative to apps dir and the mcuboot case relative to conf file dir / mcuboot dir is starting to make configs inconsistent, and hence will be confusing to users.

So until this is settled, then I would prefer to keep relative path out of the solution.